packages/mongodb-chatbot-server/src/routes/conversations/commentMessage.ts

```diff
@@ -0,0 +1,158 @@
+import { ObjectId } from "mongodb-rag-core";
+import { ConversationsService } from "../../services/ConversationsService";
+import {
+  Request as ExpressRequest,
+  Response as ExpressResponse,
+  NextFunction,
+} from "express";
+import { getRequestId, logRequest, sendErrorResponse } from "../../utils";
+import { z } from "zod";
+import { SomeExpressRequest } from "../../middleware/validateRequestSchema";
+
+export type CommentMessageRequest = z.infer<typeof CommentMessageRequest>;
+
+export const CommentMessageRequest = SomeExpressRequest.merge(
+  z.object({
+    headers: z.object({
+      "req-id": z.string(),
+    }),
+    params: z.object({
+      conversationId: z.string(),
+      messageId: z.string(),
+    }),
+    body: z.object({
+      comment: z.string().min(1, { message: "Comment cannot be empty" }),
+    }),
+  })
+);
+
+export interface CommentMessageRouteParams {
+  conversations: ConversationsService;
+  maxCommentLength?: number;
+}
+
+export function makeCommentMessageRoute({
+  conversations,
+  maxCommentLength,
+}: CommentMessageRouteParams) {
+  return async (
+    req: ExpressRequest,
+    res: ExpressResponse<void>,
+    next: NextFunction
+  ) => {
+    const reqId = getRequestId(req);
+    try {
+      const { conversationId: conversationIdStr, messageId: messageIdStr } =
+        req.params;
+      const { comment } = req.body;
+      let conversationId: ObjectId, messageId: ObjectId;
+      try {
+        conversationId = new ObjectId(conversationIdStr);
+      } catch (err) {
+        return sendErrorResponse({
+          reqId,
+          res,
+          httpStatus: 400,
+          errorMessage: "Invalid conversation ID",
+        });
+      }
+      try {
+        messageId = new ObjectId(messageIdStr);
+      } catch (err) {
+        return sendErrorResponse({
+          reqId,
+          res,
+          httpStatus: 400,
+          errorMessage: "Invalid message ID",
+        });
+      }
+
+      if (maxCommentLength && comment.length > maxCommentLength) {
+        return sendErrorResponse({
+          reqId,
+          res,
+          httpStatus: 400,
+          errorMessage: `Comment must contain ${maxCommentLength} characters or fewer`,
+        });
+      }
+
+      const conversationInDb = await conversations.findById({
+        _id: conversationId,
+      });
+
+      if (!conversationInDb) {
+        return sendErrorResponse({
+          reqId,
+          res,
+          httpStatus: 404,
+          errorMessage: "Conversation not found",
+        });
+      }
+
+      const existingMessage = conversationInDb.messages.find((message) =>
```

Comment:
since user's will probably be commenting on msgs near the end of the comment array, consider using [`Array.prototype.findLast()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findLast) instead of `Array.prototype.find()`.